### DISCUSSION TITLE
Is there a way to enable tracking of a variable belong to a class in python using pyo3?

### DISCUSSION THREAD

**Discussion by jymchng** [PyO3/pyo3#3030] [General]
```
class A:

    def __init__(self, a):
         self._a = a
    
    @decorator 
    def a(self):
        return self._a
```

The decorator can call the wrapped `a` and get the value of `self._a`.

What if I don't want the value of `self._a` but I want the pointer to `self._a`, such that whatever value `self._a` changed to, the pointer to `self._a` will always return me the newly updated value of `self._a`.

Thank you.

**Comment by adamreichold** [comment]
> What if I don't want the value of self._a but I want the pointer to self._a, such that whatever value self._a changed to, the pointer to self._a will always return me the newly updated value of self._a.

Then you need to pass around `self` and access its `_a` field via `getattr`. Because e.g. setting to `_a` to different string will make it point to a different object, was stays the same but is modified is the enclosing object referenced by `self` in your example.

Alternatively, can store you can store a reference to the `a` method (also obtain via `getattr`) and invoke `call0` on it to retrieve the value which would make the Rust code generic over whether an attribute or something else is accessed.

**Reply by jymchng** [reply]
Thank you, in that case, would that mean that for any method of any class that the decorator may wrap, there is a method in pyo3 to get that name of the variable assigned to the class?

**Reply by jymchng** [reply]
e.g. in my example, can a decorator get the name of the returned variable, i.e. `_a`?

**Reply by adamreichold** [reply]
> e.g. in my example, can a decorator get the name of the returned variable, i.e. _a?

I have no idea. (But I don't think it is specific to PyO3.)

You could probably try to parse the methods byte count in the decorator implementation, but that does not really sound viable to me as a method could implement an arbitrarily complex method to produce its return value.

**Reply by jymchng** [reply]
Thank you for replying, if that's the case then I shall give up on persuing this line and do other things then! lol
